File: ece3822code/lab_02/student.py
from statistics import mean
class student:
    # Grades Vector
    grades=[]
    
    # Default Constructor
    def __init__(self):
        pass

    # Name Set Function
    def set_name(self,name):
        self.name=name

    # Adding grades to grades vector
    def add_grade(self,grade):
        self.grades.append(grade)

    # Showing the highest grade of the student's grades
    def get_max(self):
        return(max(self.grades))

    # Showing the mean of the student's grades
    def get_average(self):
        return(mean(self.grades))

    # Displaying the student's name and their grades
    def display(self):
        print(self.name,end="")
        print(":",end=" ")
        print(*self.grades,sep = ", ")
     
    

File: lab_02/student.cpp
// including headerfile
#include "student.h"

// default constructor
student::student(){
  name="John/Jane Doe";
  numofgrades=-1;
}

// set name function
void student::set_name(string inputname){
  name=inputname;
}
l
// adding grades into a matrix
void student::add_grade(int inputgrade){
  numofgrades++;
  grades.push_back(inputgrade);
}

// find the max grade of student
int student::get_max(){
  int max=grades[0];
  for(int i = 1;i<=numofgrades;i++) {
    if(grades[i]>max) max=grades[i];
  }
  return(max);
}

// finding the average of the students grades
double student::get_average(){
  double average;
  double sum;
  for(int i=0;i<=numofgrades;i++) {
    sum+=grades[i];
  }
  average=sum/(numofgrades+1);
  return(average);
}

// displaying students stats up to assignment standard
void student::display() {
  cout << name << ": ";
  cout << grades[0];
  for (int i = 1; i <= numofgrades; i++){
    cout << ", " << grades[i];
  }
  cout<<"\n";
}

//default destructor
student::~student(){

}
